helios-services/src/main/java/com/spotify/helios/agent/HealthCheckerFactory.java

```diff
@@ -90,14 +93,17 @@ public boolean check(final String containerId) {
           healthCheck.getCommand().toArray(new String[healthCheck.getCommand().size()]);
 
       try {
-        final String execId = docker.execCreate(
+        final ExecCreation execCreation = docker.execCreate(
             containerId, cmd,
             DockerClient.ExecCreateParam.attachStdout(),
             DockerClient.ExecCreateParam.attachStderr());
+        final String execId = execCreation.id();
 
-        final String output;
+        String output = "";
         try (LogStream stream = docker.execStart(execId)) {
           output = stream.readFully();
+        } catch (IOException e) {
+          // thrown from implicit call to close(). Ignore.
```

Comment:
I guess we shouldn't ignore? Redefining close() seems like a better idea though.